src/metabase/server/middleware/offset_paging.clj

```diff
@@ -36,8 +42,20 @@
   [handler]
   (fn [request respond raise]
     (if-let [{:keys [limit offset] :as paging-params} (parse-paging-params request)]
-      (binding [*limit* limit
+      (binding [*limit*  limit
                 *offset* offset
                 *paged?* true]
         (handler (with-paging-params request paging-params) respond raise))
       (handler request respond raise))))
+
+(defn page-result
+  "If *offset* and *limit* are defined, returns the appropriate segment of `result`. Otherwise returns the whole thing.
+
+  If at all possible, you should implement this logic in your SQL query, but if that's not possible this function may
```

Comment:
Having this convenience provided may be too enticing for the lazy - given how hot a topic non-scalable database queries are. Can we make it a discouraged var?